aiohttp/test_utils.py

```diff
@@ -78,6 +78,8 @@ def _make_factory(self):
         pass
 
     def make_url(self, path):
+        if isinstance(path, URL):
```

Comment:
I'm not sure what `path.startswith('/')` is checking, but in your solution if we pass `http://foo.com/` as `path` it returns `http://foo.com/` for any `self._root`. Is it ok?